exercises/react/src/example.c

```diff
@@ -41,42 +43,54 @@ struct reactor *create_reactor()
    return calloc(1, sizeof(struct reactor));
 }
 
-void destroy_reactor(struct reactor *r)
+static void destroy_cell(struct cell *c)
 {
-   struct child *child = r->first;
-   while (child) {
-      struct cb *cb = child->cell->cb;
-      while (cb) {
-         struct cb *next_cb = cb->next;
-         free(cb);
-         cb = next_cb;
-      }
-      free(child->cell);
+   struct cb *cb = c->cb;
+   while (cb) {
```

Comment:
my submission on the website asks whether it is possible to reduce the duplication between the three while loops